File: app/static/passwordvalidator.js
export function validatePassword(value){
    let errors = "";

    if(value.length < 10) {
        errors += "&#8226; Must be at least 10 characters.<br>";
    }

    if(value.length >= 50) {
        errors += "&#8226; Must be less than 50 characters.<br>";
    }

    if(!value.match(/[A-Z]{1,}/)) {
        errors += "&#8226; Must contain at least one capital letter.<br>";
    }

    if(!value.match(/[0-9]{1,}/)) {
        errors += "&#8226; Must contain at least one number.<br>";
    }

    if(!value.match(/[~`!@#$%^&*()\-_+={}[\]|\\\;:"<>,./?]{1,}/)) {
        errors += "&#8226; Must contain at least one special character.<br>";
    }

    return errors;
}

export function confirmPassword(original, confirm){
    if(original.length < 1 || confirm.length < 1) {
        return false;
    } 
    return original != confirm;
}

File: app/password_validator.py
from wtforms.validators import ValidationError

"""
ADAPTED FROM MY COURSEWORK ONE SUBMISSION

Flask WTForms validator for password fields.
Ensures a date is between a minimum and maximum date.

Follows the structure defined in other Flask Validators.
https://wtforms.readthedocs.io/en/2.3.x/validators/
"""
class PasswordValidator(object):
    def __init__(self, optional=False):
        self.optional = optional

    def __call__(self, form, field):
        self.validate_password(field.data)                                                                                                                                                          

    """
    Validation method for the Password.
    Ensures password meets requirements.
    """
    def validate_password(self, value):
        errors = []

        if self.optional and len(value) < 1:
            return

        if len(value) < 10:
            errors.append("Must be at least 10 characters.")

        if len(value) >= 50:
            errors.append("Must be less than 50 characters.")

        if not any(char.isupper() for char in value):
            errors.append("Must contain at least one capital letter.")

        if not any(char.isdigit() for char in value):
            errors.append("Must contain at least one number.")

        if not any(char in "~`!@#$%^&*()-_+={}[]|\\;:'\"<>,./?" for char in value):
            errors.append("Must contain at least one special character.")

        if len(errors) > 0:
            raise ValidationError(errors)